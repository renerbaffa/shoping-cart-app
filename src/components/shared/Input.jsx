import React from 'react';

/*
  Input component is an adapter.
  Imagine that in the future we move to react-bootstrap or material-ui
  In this case we only need to replase default input by the library input
*/
const Input = (props) => (
  <input {...props} />
);

export default Input;
