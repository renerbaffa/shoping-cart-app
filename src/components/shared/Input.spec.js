import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';

describe('<Input />', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Input className="className" value="value" placeholder="Search" value="Search" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
