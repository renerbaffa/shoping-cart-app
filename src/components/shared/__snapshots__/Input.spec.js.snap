// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly 1`] = `
<input
  className="className"
  placeholder="Search"
  value="Search"
/>
`;
